tests/tests_data/streaming/test_reader.py

```diff
@@ -95,8 +70,34 @@ def test_prepare_chunks_thread(tmpdir):
 
     cache._reader._try_load_config()
 
-    thread = PrepareChunksThread(cache._reader.config, item_loader=PyTreeLoader(), max_cache_size=1)
-    assert thread._delete_chunks_when_processed
+    assert len(os.listdir(cache_dir)) == 14
 
     thread = PrepareChunksThread(cache._reader.config, item_loader=PyTreeLoader(), max_cache_size=10000)
     assert not thread._delete_chunks_when_processed
+
+    thread = PrepareChunksThread(cache._reader.config, item_loader=PyTreeLoader(), max_cache_size=1)
+    assert thread._delete_chunks_when_processed
+
+    thread.start()
+
+    assert thread._pre_download_counter == 0
+
+    thread.download([0, 1, 2, 3, 4, 5, _END_TOKEN])
+
+    sleep(0.25)
```

Comment:
These sleeps can be a source of flakiness. Can you replace them with a barrier that ensures we have downloaded before moving on? We should not rely on this sleep if the task needs to complete in a thread.

For example, you could wait here until `thread._pre_download_counter > 0`, and then run the assertions.

Reply:
done